### PR TITLE
Major updates to GNN example

### PR DESCRIPTION
**Pull request by ZiyueXu77**
Fixes # .

### Description

Split dataset to clients, rewrite code for training following latest API, add financial data example

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Quick tests passed locally by running `./runtest.sh`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated.

### CODE REVIEW

File: examples/advanced/gnn/code/pyg_sage.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import torch.nn as nn
```
Comment (YuanTingHsieh) [inline examples/advanced/gnn/code/pyg_sage.py:1]: Please add license header to this file.

If this file is adapted from other sources, we can cite it as well.

File: examples/advanced/gnn/jobs/gnn_finance/app_1/custom/pyg_sage.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
+import torch.nn as nn
```
Comment (YuanTingHsieh) [inline examples/advanced/gnn/jobs/gnn_finance/app_1/custom/pyg_sage.py:1]: need to add license header to this file

File: examples/advanced/gnn/jobs/gnn_finance/app_2/custom/pyg_sage.py [hunk 3]
```diff
@@ -0,0 +1,23 @@
+import torch.nn as nn
```
Comment (YuanTingHsieh) [inline examples/advanced/gnn/jobs/gnn_finance/app_2/custom/pyg_sage.py:1]: need to add license header to this file

File: examples/advanced/gnn/jobs/gnn_finance/app_server/custom/pyg_sage.py [hunk 4]
```diff
@@ -0,0 +1,23 @@
+import torch.nn as nn
```
Comment (YuanTingHsieh) [inline examples/advanced/gnn/jobs/gnn_finance/app_server/custom/pyg_sage.py:1]: need to add license header to this file

**Review by YuanTingHsieh — COMMENTED** [review]
Great example.

If you could, please create a job_template "gnn_pt" and then just generate these jobs using the same template.
If you don't know 100% how to do that, I can do that in a follow up PR

File: examples/advanced/gnn/code/pyg_sage.py (hunk 1 above)
Comment (ZiyueXu77) [reply examples/advanced/gnn/code/pyg_sage.py:15]: thanks, added

**Review by YuanTingHsieh — APPROVED** [review]
Thanks, LGTM!

**Pull request merged**